### PR TITLE
8273092: Sort classlist in JDK image

### PR DESCRIPTION
**Pull request by iklam**
When the classlist is generated using build.tools.classlist.HelloClasslist, its contents may be non-deterministic due to Java thread execution order.

We should sort the generated classlist to make the JDK image's contents more deterministic.

Tested with Mach5 tier1, tier2, builds-tier5

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8273092](https://bugs.openjdk.java.net/browse/JDK-8273092): Sort classlist in JDK image

### Reviewers
 * [Claes Redestad](https://openjdk.java.net/census#redestad) (@cl4es - **Reviewer**) ⚠️ Review applies to dc170ec082b835fd4d4387788ad66de98d84ddcb
 * [Magnus Ihse Bursie](https://openjdk.java.net/census#ihse) (@magicus - **Reviewer**) ⚠️ Review applies to dc170ec082b835fd4d4387788ad66de98d84ddcb
 * [Daniel Fuchs](https://openjdk.java.net/census#dfuchs) (@dfuch - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5288/head:pull/5288` \
`$ git checkout pull/5288`

Update a local copy of the PR: \
`$ git checkout pull/5288` \
`$ git pull https://git.openjdk.java.net/jdk pull/5288/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5288`

View PR using the GUI difftool: \
`$ git pr show -t 5288`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5288.diff">https://git.openjdk.java.net/jdk/pull/5288.diff</a>

</details>

### CODE REVIEW

File: make/jdk/src/classes/build/tools/classlist/SortClasslist.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * This application is meant to be run to create a classlist file representing
+ * common use.
+ *
+ * The classlist is produced by adding -XX:DumpLoadedClassList=classlist
+ */
```
Comment (cl4es) [inline make/jdk/src/classes/build/tools/classlist/SortClasslist.java:31]: Looks like a copy-paste error (also redundant)

**Review by cl4es — APPROVED** [review]
Seems OK.

**Review by magicus — APPROVED** [review]
Looks great. Thank you for helping to make the build deterministic!

File: make/jdk/src/classes/build/tools/classlist/SortClasslist.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        ArrayList<String> classes = new ArrayList<>();
+        ArrayList<String> lambdas = new ArrayList<>();
+
+        FileInputStream fis = new FileInputStream(args[0]);
+        Scanner scanner = new Scanner(fis);
+        Pattern p = Pattern.compile("^(.*)[ ]+id:[ ]+([0-9]+)$");
+        while (scanner.hasNextLine()) {
+            String line = scanner.nextLine();
+            Matcher m = p.matcher(line);
+            if (m.find()) {
```
Comment (dfuch) [inline make/jdk/src/classes/build/tools/classlist/SortClasslist.java:58]: Are we sure that a comment line will not match this regexp, or that if it matches, it is not a comment line?

Comment (iklam) [reply make/jdk/src/classes/build/tools/classlist/SortClasslist.java]: Thanks for the comments. I've swapper the matching order to check for leading `#` and `@` characters first.

**Review by dfuch — APPROVED** [review]
Thanks Ioi! These changes look good to me.

**Comment by iklam** [comment]
Thanks @cl4es @magicus @dfuch for the review!
/integrate

**Pull request closed without merging**